### DISCUSSION TITLE
Does MAUI support file drag and drop to other application (Windows, Linux, macOS)?

### DISCUSSION THREAD

**Discussion by HubKing** [dotnet/maui#7508] [Q&A]
For example, you can drag a file from File Explorer and drop it to Notepad. In WinForms, you can achieve that with two lines of code:

    var data = new DataObject(DataFormats.FileDrop, new[]{"C:\\sample.txt"});
    control.DoDragDrop(data, DragDropEffects.Copy);

Can it be done with the current MAUI for desktop OSes (Windows, Linux, macOS)? If not, is there a plan to support it? Or since MAUI is sort of mobile-first platform, it has no such feature and has no plan to support this?

**Comment by carlosRamirezVel** [comment]
Im looking for answer to the sample issue, did you find any solution or work around?

**Reply by HubKing** [reply]
No. As you see, there is not even a reply from any Microsoft developers, from which I could safely assume that it has no such feature. Probably all this is because MAUI is not really a desktop GUI framework, but some stupid mobile GUI framework that can also run on desktop. That is why it hardly supports any desktop-specific features. 

My guess is that this MAUI will be another big failure of Microsoft, just like Windows Phone. There are already tons of mobile-first cross-platform GUI frameworks. Why would mobile developers want another such framework? The only way it could excel others would be providing first-class desktop support, but these MS developers forgot what their root is and are trying to pursue what others do.

**Reply by akhanalcs** [reply]
@HubKing You're exactly right.
Microsoft is a follower now. They don't know how to lead and innovate these days.

**Comment by SunshineSpring666** [comment]
Hi, everyone. 
Expecting such feature of dragging a file from outside the app (e.g., from file explorer) into the app, and trigger the OnDrop event, so as to process the file's data.

Currently, dragging a file into the visual element which set DropGestureRecognizer causes UnhandledException with message: 
Object reference not set to an instance of an object. Value cannot be null. (Parameter 'view')

Any clues or workarounds?

Thanks a lot.

**Reply by marcojak** [reply]
> Hi, everyone. Expecting such feature of dragging a file from outside the app (e.g., from file explorer) into the app, and trigger the OnDrop event, so as to process the file's data.
> 
> Currently, dragging a file into the visual element which set DropGestureRecognizer causes UnhandledException with message: Object reference not set to an instance of an object. Value cannot be null. (Parameter 'view')
> 
> Any clues or workarounds?
> 
> Thanks a lot.

Getting the same error. It would be good to know if this is or will be possible.

Thanks

**Comment by upswing1** [comment]
I am looking for an answer to this as well.

**Comment by VladislavAntonyuk** [comment]
https://vladislavantonyuk.github.io/articles/Drag-and-Drop-any-content-to-a-.NET-MAUI-application
